src/main/java/com/telefonica/iot/cygnus/backends/mongo/MongoBackend.java

```diff
@@ -375,9 +376,52 @@ private BasicDBList buildPrepopulatedPoints(Resolution resolution) {
      * @param mds
      * @throws Exception
      */
-    public void insertContextDataRaw(String dbName, String collectionName, String recvTime,
-            Map<String, String> attrs, Map<String, String> mds) throws Exception {
-        // FIXME: the row-like column insertion mode is not currently available for Mongo
+    public void insertContextDataRaw(String dbName, String collectionName, long recvTimeTs, String recvTime,
+                                     String entityId, String entityType, Map<String, String> attrs,
+                                     Map<String, String> mds) throws Exception {
+
+        MongoDatabase db = getDatabase(dbName);
+        MongoCollection collection = db.getCollection(collectionName);
+        Document doc = new Document("recvTime", new Date(recvTimeTs * 1000));
+
+        switch (dataModel) {
+            case COLLECTIONPERSERVICEPATH:
+                doc.append("entityId", entityId)
+                        .append("entityType", entityType);
+
+                Iterator it = attrs.entrySet().iterator();
+                while (it.hasNext()) {
+                    Map.Entry pair = (Map.Entry) it.next();
+                    String attrName = (String) pair.getKey();
+                    String attrValue = (String) pair.getValue();
+                    doc.append(attrName, attrValue);
+                }
+
+                break;
+            case COLLECTIONPERENTITY:
+                it = attrs.entrySet().iterator();
+                while (it.hasNext()) {
+                    Map.Entry pair = (Map.Entry) it.next();
+                    String attrName = (String) pair.getKey();
+                    String attrValue = (String) pair.getValue();
+                    doc.append(attrName, attrValue);
+                }
+                break;
+            case COLLECTIONPERATTRIBUTE:
```

Comment:
In fact, this data model has no sense when persisting the data in column-like mode. In this case I think a warning log must be written and nothing else must be done regarding the persistence.